prestoadmin/prestoclient.py

```diff
@@ -191,3 +217,82 @@ def get_rows(self, num_of_rows=NUM_ROWS):
 
     def get_next_uri(self):
         return self.next_uri
+
+    def _get_connection(self):
+        if self.coordinator_config.use_https():
+            return self._get_https_connection()
+        else:
+            return HTTPConnection(self.server, self.port, False, URL_TIMEOUT_MS)
+
+    def _get_https_connection(self):
+        ca_file_path = self._get_pem()
+        https_port = self.coordinator_config.get_https_port()
+        result = HTTPSCaCertConnection(
+                self.server, https_port, None, None, ca_file_path, False, URL_TIMEOUT_MS)
+        return result
+
+    def _get_local_keystore_path(self):
+        remote_keystore_path = self.coordinator_config.get_client_keystore_path()
+        keystore_name = os.path.basename(remote_keystore_path)
+        return os.path.join(get_coordinator_directory(), keystore_name)
+
+    def _fetch_keystore_data(self):
+        if not self.keystore_data:
+            remote_keystore_path = self.coordinator_config.get_client_keystore_path()
+            keystore_data = StringIO()
+            get(remote_keystore_path, keystore_data, use_sudo=True)
```

Comment:
Useful errors now happen in both cases; presto_config calls `error()` if a key is missing, and fabric.get shows an error if the file is missing.